test/util/discovery/discovery.go

```diff
@@ -0,0 +1,72 @@
+package discovery
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	sriovv1 "github.com/openshift/sriov-network-operator/pkg/apis/sriovnetwork/v1"
+	"github.com/openshift/sriov-network-operator/test/util/client"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Enabled indicates whether test discovery mode is enabled.
+func Enabled() bool {
+	discoveryMode, _ := strconv.ParseBool(os.Getenv("DISCOVERY_MODE"))
+	return discoveryMode
+}
+
+func getSriovNodes(clients *client.ClientSet, sriovNodes []string) ([]corev1.Node, error) {
+	allNodes, err := clients.Nodes().List(context.Background(), metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker",
+	})
+	if err != nil {
+		return nil, err
+	}
+	var nodes []corev1.Node
+	for _, node := range allNodes.Items {
+		if containsNode(node.ObjectMeta.Name, sriovNodes) {
+			nodes = append(nodes, node)
+		}
+	}
+	return nodes, nil
+}
+
+func containsNode(name string, sriovNodes []string) bool {
```

Comment:
In any case, can you put this and all other "util functions" below `DiscoverResources` ? As they are secondary and used by `DiscoverResources`, it's fine to find them after in reading order.

Reply:
Done